x-pack/plugins/watcher/public/models/action/jira.action.js

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { get } from 'lodash';
+import { BaseAction } from './base_action';
+import { i18n } from '@kbn/i18n';
+
+const requiredFields = ['fields'];
+const optionalFields = ['account', 'proxy'];
+
+const allFields = [...requiredFields, ...optionalFields];
+
+export class JiraAction extends BaseAction {
+  constructor(props = {}) {
+    super(props);
+
+    this.fields = {};
+    allFields.forEach((field) => {
+      this.fields[field] = get(props, field);
+    });
+  }
+
+  get upstreamJson() {
+    // Add all required fields to the request body
+    let result = requiredFields.reduce((acc, field) => {
+      acc[field] = this.fields[field];
+      return acc;
+    }, super.upstreamJson);
+
+    // If optional fields have been set, add them to the body
+    result = optionalFields.reduce((acc, field) => {
+      if (this[field]) {
+        acc[field] = this.fields[field];
+      }
+      return acc;
+    }, result);
+
+    return result;
+  }
+
+  get description() {
+    return i18n.translate('xpack.watcher.models.jiraAction.description', {
+      defaultMessage: '{issueName} will be created in Jira',
+      values: {
+        issueName: this.fields && this.fields.issuetype && this.fields.issuetype.name || ''
```

Comment:
From the [Jira action docs](https://www.elastic.co/guide/en/x-pack/current/actions-jira.html) it looks like the expected shape is:

```json
{
  "account" : "integration-account", 
  "fields" : {
    "issue" : {
      "issuetype" : {
        "name": "Bug" 
      }
    }
  }
}
```

So I think the correct path should be `this.fields.fields.issue.issuetype.name`, right?

If we're already using lodash's `get`, I think we might as well use that here too:

```js
get(this.fields, 'fields.issue.issuetype.name', '');
```

Reply:
Good catch!